build.sh

```diff
@@ -2,13 +2,32 @@
 # the path to the ComputeCPP package root directory, e.g. /home/user/ComputeCpp-CE-0.1-Linux/
 PACKAGE_ROOT=$1
 
-function install_gmock  {
-  mkdir -p external && pushd external
-  git clone git@github.com:google/googletest.git
-  pushd googletest/googlemock/make && make
-  popd 
-  popd
-}
+#compute the number of cores
+NPROC=$(nproc)
+
+function install_gmock  {(
+  REPO="git@github.com:google/googletest.git"
+  #REPO="https://github.com/google/googletest.git"
+  if [ -d external ]
+  then
+    cd external
+    if [ -d googletest ]
+    then
+      cd googletest
+      git pull
+    else
+      git clone $REPO
+      cd googletest
+    fi
+  else
+    mkdir external
+    cd external
+    git clone $REPO
```

Comment:
I have the feeling the logic could be simplified a little bit here to avoid having the git clone command used twice. What about using BDD to minimize the code size here? :-)
Feel free to add comments.